modules/data-entry/src/main/frontend/src/dataHomepage/ExportButton.jsx

```diff
@@ -0,0 +1,395 @@
+    },
+  },
+  entryActionIcon: {
+    float: "right",
+    marginRight: theme.spacing(1),
+  },
+  variableDropdown: {
+    "& > .MuiInputLabel-root" : {
+      maxWidth: `calc(100% - ${theme.spacing(3)})`,
+    },
+  },
+  variableOption: {
+    whiteSpace: "normal",
+  },
+  valueEntry: {
+    border: "1px solid " + theme.palette.divider,
+    borderRadius: theme.spacing(.5, 3, 3, .5),
+    margin: theme.spacing(1, 0),
+    "& > .MuiGrid-item" : {
+      display: "flex",
+      paddingLeft: theme.spacing(1.5),
+      alignItems: "center",
+    },
+  },
+  valueActions: {
+    justifyContent: "flex-end",
+    paddingTop: "0!important"
+  },
+  avatar: {
+    marginRight: theme.spacing(1),
+    marginTop:  theme.spacing(1),
+    alignSelf: "start",
+    zoom: "75%"
+  }
+}));
+
+let findQuestionsOrSections = (json, result = []) =>  {
+  Object.entries(json || {}).forEach(([k,e]) => {
+    if (e?.['jcr:primaryType'] == "cards:Question" || e?.['jcr:primaryType'] == "cards:Section") {
+      result.push({name: e['@name'], text: e['text'] || e['label'], path: e['@path'], type: e['jcr:primaryType'].replace("cards:", '')});
+      e?.['jcr:primaryType'] == "cards:Section" && findQuestionsOrSections(e, result);
+    } else if (typeof(e) == 'object') {
+      findQuestionsOrSections(e, result);
+    }
+  })
+  return result;
+}
+
+let entitySpecs = {
+  Question: {
+    avatarColor: deepPurple[700]
+  },
+  Section: {
+    avatar: "view_stream",
+    avatarColor: orange[800]
+  }
+}
+
+/**
+ * A component that renders an icon or button to open the export dialog that generates an export URL for an entry.
+ */
+function ExportButton(props) {
+  const { entityData, entryLabel, entryPath, entryName, entryType, variant, size, onClose } = props;
+
+  const [ open, setOpen ] = useState(false);
+  // List of questions and sections to display in dropdown select to exclude/include
+  const [ entities, setEntities] = useState();
+
+  // Decides if the generated export URL ends in .csv or in .tsv
+  const [ fileFormat, setFileFormat ] = useState(".csv");
+  // Decides if/how .csvHeader is specified:
+  // by default we have csvHeader:labels
+  // to disable labels, add .-csvHeader:labels
+  // to enable identifiers, add .csvHeader:raw
+  const [ labelFormat ,setLabelFormat ] = useState(true);
+  const [ identifiersFormat, setIdentifiersFormat ] = useState(false);
+  // Specifies if the .labels processor is enabled (disabled by default for values)
+  const [ hasLabeles, setHasLabeles ] = useState(false);
+
+  // Column selection:
+  // just one of the Include or Exclude options should be available at a time
+  const [ isInclude, setIsInclude ] = useState(false);
+  // List of quesitons ids to Include OR Exclude
+  const [ questionIds, setQuestionIds ] = useState([]);
+  const [ tempValue, setTempValue ] = useState(''); // Holds new, non-selected values
+
+  const buttonText = entryLabel || ("Export " + (entryType?.toLowerCase() || '')).trim();
```

Comment:
Since this button is specifically for exporting forms for a certain questionnaire and doesn't need to have a generic text based on the entity type, we could just change this to
```suggestion
  const buttonText = entryLabel || "Export forms";
```
This will also help have the correct tooltip in the Questionnaires table.